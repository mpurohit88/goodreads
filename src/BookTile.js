import React from 'react';
import PropTypes from 'prop-types';
import { withStyles } from '@material-ui/core/styles';
import Grid from '@material-ui/core/Grid';
import Paper from '@material-ui/core/Paper';
import Typography from '@material-ui/core/Typography';
import ButtonBase from '@material-ui/core/ButtonBase';

const styles = theme => ({
  root: {
    flexGrow: 1,
    maxWidth: 700,
    padding: theme.spacing.unit * 2,
    'margin-top':2,
    'margin-right':'auto',
    'margin-bottom':0,
    'margin-left':'auto',
    'text-align': 'left',
    cursor: 'pointer'
  },
  image: {
    width: 100,
    height: 100,
  },
  img: {
    margin: 'auto',
    display: 'block',
    maxWidth: '100%',
    maxHeight: '100%',
  },
});

function Tile(props) {
  const { classes, searchResult, getBookDetails, index } = props;
  
  return (
    <Paper className={classes.root} key={'id-' + index} onClick={() => getBookDetails(searchResult.best_book.id["#text"])}>
      <Grid container spacing={16}>
        <Grid item>
          <ButtonBase className={classes.image}>
            <img className={classes.img} alt="complex" src={searchResult.best_book.small_image_url} />
          </ButtonBase>
        </Grid>
        <Grid item xs={12} sm container>
          <Grid item xs container direction="column" spacing={16}>
            <Grid item xs>
              <Typography gutterBottom variant="h5">
                {searchResult.best_book.title}
              </Typography>
            </Grid>
          </Grid>
        </Grid>
      </Grid>
    </Paper>
  );
}

Tile.propTypes = {
  classes: PropTypes.object.isRequired,
};

export default withStyles(styles)(Tile);
