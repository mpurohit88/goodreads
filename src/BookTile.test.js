import React from 'react';
import ReactDOM from 'react-dom';
import BookTile from './BookTile';
import { shallow } from 'enzyme';

import { } from './setupTests';

it('renders without crashing', () => {
  const searchResult = {text_reviews_count: {"#text": "324"}, ratings_count: {"#text": "23"}, average_rating: 123,
                        best_book: {sort_image_url: 'https://www.goodreads.com/img.png', author: {name: "React"}}}
  const div = document.createElement('div');
  ReactDOM.render(<BookTile searchResult={searchResult} index={1}/>, div);
  ReactDOM.unmountComponentAtNode(div);
});

it('render Book Tile correctly', () => {
  const wrapper = shallow(<BookTile />);
  expect(wrapper).toMatchSnapshot();
});