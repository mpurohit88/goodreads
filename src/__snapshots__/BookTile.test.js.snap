// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render Book Tile correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <WithStyles(Tile) />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "classes": Object {
        "image": "Tile-image-174",
        "img": "Tile-img-175",
        "root": "Tile-root-173",
      },
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "classes": Object {
          "image": "Tile-image-174",
          "img": "Tile-img-175",
          "root": "Tile-root-173",
        },
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
